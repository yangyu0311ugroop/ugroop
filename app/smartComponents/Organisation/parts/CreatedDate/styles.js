const styles = () => ({
  lineIndicator: {
    width: 40,
    height: 2,
    backgroundColor: '#8abd84',
  },
  name: {
    lineHeight: 1.6,
    marginTop: 8,
  },
  createddate: {
    fontSize: 14,
    padding: '8px 0 8px 0',
  },
});

export default styles;
