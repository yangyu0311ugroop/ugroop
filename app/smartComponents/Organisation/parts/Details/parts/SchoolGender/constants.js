const TITLE = 'schoolGender';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};
export const ORG_FORM_NAME = 'type';

export const SCHOOL_GENDERS = [
  { code: 'boys', name: 'Boys School' },
  { code: 'girls', name: 'Girls School' },
  { code: 'coEducational', name: 'Co-Educational' },
];
