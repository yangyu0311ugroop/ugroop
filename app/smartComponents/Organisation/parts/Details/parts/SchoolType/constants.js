const TITLE = 'schooltype';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};
export const ORG_FORM_NAME = 'schooltype';

export const SCHOOL_TYPES = [
  'Primary',
  'Secondary',
  'K-10/12',
  'Special',
  'College',
  'University',
];
