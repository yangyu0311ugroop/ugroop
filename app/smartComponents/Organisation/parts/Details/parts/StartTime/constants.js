const TITLE = 'starttime';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'starttime';
