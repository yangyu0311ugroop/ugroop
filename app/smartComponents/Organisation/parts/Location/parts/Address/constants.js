const TITLE = 'address';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'address';
