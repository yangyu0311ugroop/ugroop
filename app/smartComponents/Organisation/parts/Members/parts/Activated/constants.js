const TITLE = 'activated';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'activated';
