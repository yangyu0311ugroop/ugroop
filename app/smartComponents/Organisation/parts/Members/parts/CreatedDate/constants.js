const TITLE = 'createdAt';

export const ORG_FIELD_VARIANTS = {
  TITLE,
};

export const ORG_FORM_NAME = 'createdAt';
