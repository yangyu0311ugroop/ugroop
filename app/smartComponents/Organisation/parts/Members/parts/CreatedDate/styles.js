const styles = () => ({
  lineIndicator: {
    width: 40,
    height: 2,
    backgroundColor: '#8abd84',
  },
  name: {
    lineHeight: 1.6,
    marginTop: 8,
  },
  createddate: {
    fontSize: 14,
    padding: '8px 0 8px 0',
  },
  createdDateFontMobile: {
    fontSize: 12,
  },
  memberSince: {
    fontWeight: 500,
  },
  noUnderline: {
    borderBottom: 'none !important',
    cursor: 'inherit !important',
    textDecoration: 'none !important',
  },
});

export default styles;
