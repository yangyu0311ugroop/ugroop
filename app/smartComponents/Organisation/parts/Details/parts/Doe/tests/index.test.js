import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import React from 'react';
import { ORG_FIELD_VARIANTS } from 'smartComponents/Organisation/constants';
import mockStylesheet from 'utils/mockStylesheet';
import theme from 'theme/coolTheme';
import styles from '../styles';
import { Doe } from '../index';

const mockStyles = mockStylesheet('Website', styles, theme);

describe('<Doe />', () => {
  let rendered;
  let instance;

  const resaga = {
    setValue: jest.fn(),
    dispatchTo: jest.fn(),
  };

  const intl = {
    formatMessage: jest.fn(),
  };

  const props = {
    classes: mockStyles,
    resaga,
    intl,
  };

  beforeEach(() => {
    rendered = shallow(<Doe {...props} />);
    instance = rendered.instance();
    jest.clearAllMocks();
  });

  it('should exists', () => {
    expect(Doe).toBeDefined();
  });

  it('should render without exploding', () => {
    expect(rendered.length).toBe(1);
  });

  describe('render()', () => {
    it('should render correctly', () => {
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text only variant', () => {
      rendered.setProps({
        variant: ORG_FIELD_VARIANTS.TEXT_ONLY,
      });
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
    it('should render text field if variant is text field', () => {
      rendered.setProps({
        variant: ORG_FIELD_VARIANTS.TEXT_FIELD,
      });
      const snapshot = shallow(<div>{instance.render()}</div>);

      expect(toJSON(snapshot)).toMatchSnapshot();
    });
  });
});
