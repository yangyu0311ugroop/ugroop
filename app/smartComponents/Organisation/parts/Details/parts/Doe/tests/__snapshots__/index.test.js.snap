// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Doe /> render() should render correctly 1`] = `
<div>
  <P
    doe=""
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
  />
</div>
`;

exports[`<Doe /> render() should render text field if variant is text field 1`] = `
<div>
  <FormsyInterceptor
    doe=""
    intl={
      Object {
        "formatMessage": [MockFunction] {
          "calls": Array [
            Array [
              Object {
                "defaultMessage": "DOE Number",
                "id": "app.smartComponents.Organisation.parts.doe.label",
              },
            ],
            Array [
              Object {
                "defaultMessage": "DOE Number",
                "id": "app.smartComponents.Organisation.parts.doe.label",
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
      }
    }
    name="doe"
    onInterceptValue={[Function]}
    value=""
  />
</div>
`;

exports[`<Doe /> render() should render text only variant 1`] = `
<div>
  <P
    doe=""
    intl={
      Object {
        "formatMessage": [MockFunction],
      }
    }
  />
</div>
`;
